test/ListProxyTest.js

```diff
@@ -0,0 +1,266 @@
+            }).then(function (all) {
+                expect(all).to.deep.equal([1, 5, 2, 3]);
+            });
+        })
+    });
+
+    it("clears", function () {
+        return listInstance.addAll([1, 2, 3]).then(function () {
+            return listInstance.clear();
+        }).then(function () {
+            return listInstance.size().then(function (size) {
+                expect(size).to.equal(0);
+            });
+        });
+    });
+
+
+    it("inserts all elements of array at index", function () {
+        return listInstance.addAll([1, 2, 3]).then(function () {
+            return listInstance.addAllAt(1, [5, 6]);
+        }).then(function () {
+            return listInstance.toArray();
+        }).then(function (all) {
+            expect(all).to.deep.equal([1, 5, 6, 2, 3])
+        });
+    });
+
+
+    it("gets item at index", function () {
+        var input = [1, 2, 3];
+        return listInstance.addAll(input).then(function () {
+            return listInstance.get(1).then(function (result) {
+                expect(result).to.equal(2);
+            });
+        });
+    });
+
+    it("removes item at index", function () {
+        var input = [1, 2, 3];
+        return listInstance.addAll(input).then(function () {
+            return listInstance.removeAt(1).then(function (replaced) {
+                expect(replaced).to.equal(2);
+            });
+        }).then(function () {
+            return listInstance.toArray().then(function (all) {
+                expect(all).to.deep.equal([1, 3]);
+            });
+        });
+    });
+
+    it("replaces item at index", function () {
+        var input = [1, 2, 3];
+        return listInstance.addAll(input).then(function () {
+            return listInstance.set(1, 6).then(function (replaced) {
+                expect(replaced).to.equal(2);
+            });
+        }).then(function () {
+            return listInstance.toArray().then(function (all) {
+                expect(all).to.deep.equal([1, 6, 3]);
+            });
+        });
+    });
+
+
+    it("contains", function () {
+        var input = [1, 2, 3];
+        return listInstance.addAll(input).then(function () {
+            return listInstance.contains(1).then(function (contains) {
+                expect(contains).to.be.true;
+            });
+        }).then(function () {
+            return listInstance.contains(5).then(function (contains) {
+                expect(contains).to.be.false;
+            });
+        });
+    });
+
+    it("contains all", function () {
+        return listInstance.addAll([1, 2, 3]).then(function () {
+            return listInstance.containsAll([1, 2]).then(function (contains) {
+                expect(contains).to.be.true;
+            });
+        }).then(function () {
+            return listInstance.containsAll([3, 4]).then(function (contains) {
+                expect(contains).to.be.false;
+            });
+        });
+    });
+
+
+    it("is empty", function () {
```

Comment:
I think this test should be divided into two different tests as it has two different scenarios.